external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriteTask.scala

```diff
@@ -113,49 +113,43 @@ private[kafka010] abstract class KafkaRowWriter(
   }
 
   private def createProjection = {
+    def expression(attrName: String)(defaultFn: () => Expression): Expression = {
+      inputSchema.find(_.name == attrName).getOrElse(defaultFn())
+    }
+
+    def assertDataType(attrName: String, desired: Seq[DataType], actual: DataType): Unit = {
+      if (!desired.exists(_.sameType(actual))) {
+        throw new IllegalStateException(s"$attrName attribute unsupported type " +
+          s"${actual.catalogString}")
+      }
+    }
+
     val topicExpression = topic.map(Literal(_)).orElse {
```

Comment:
Yeah I thought it would be complicated or require more change on expression but turned out it's not. I'll make a change like below:

```
    val topicExpression = topic.map(Literal(_)).getOrElse(
      expression(KafkaWriter.TOPIC_ATTRIBUTE_NAME) { () =>
        throw new IllegalStateException(s"topic option required when no " +
          s"'${KafkaWriter.TOPIC_ATTRIBUTE_NAME}' attribute is present")
      }
    )
```